components/batch-system/src/batch.rs

```diff
@@ -256,9 +260,14 @@ impl<N: Fsm, C: Fsm, Handler: PollHandler<N, C>> Poller<N, C, Handler> {
         }
 
         if batch.is_empty() {
-            self.handler.pause();
-            if let Ok(fsm) = self.fsm_receiver.recv() {
-                return batch.push(fsm);
+            loop {
+                if self.handler.pause() {
+                    if let Ok(fsm) = self.fsm_receiver.recv() {
+                        return batch.push(fsm);
+                    }
+                } else if let Ok(fsm) = self.fsm_receiver.recv_timeout(Duration::from_micros(100)) {
```

Comment:
I suggest to change the definition of `pause` to return `Option<Duration>` instead, which can be used for `recv_timeout`.